packages/core/http/core-http-server-internal/src/cdn.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { URL, format } from 'node:url';
+import type { CspAdditionalConfig } from './csp';
+
+export interface Input {
+  url?: string;
+}
+
+export class CdnConfig {
+  private url: undefined | URL;
+  constructor(private readonly input: Input) {
+    if (this.input.url) {
+      this.url = new URL(this.input.url); // This will throw for invalid URLs
+    }
+  }
+
+  public get host(): undefined | string {
+    return this.url?.host ?? undefined;
+  }
+
+  public get baseHref(): undefined | string {
+    if (this.url) {
+      return this.url.pathname === '/' ? this.url.origin : format(this.url);
+    }
+  }
+
+  public getCspConfig(): CspAdditionalConfig {
+    const hostname = this.host;
+    if (!hostname) return {};
+    return {
+      default_src: [hostname],
+      font_src: [hostname],
+      img_src: [hostname],
+      script_src: [hostname],
+      style_src: [hostname],
+      worker_src: [hostname],
+      connect_src: [hostname],
+      frame_src: [hostname],
```

Comment:
Have you verified that Kibana requires **all** of these declarations for the CDN to function? A few initial thoughts:

1) We don't supply a `default_src` today, and I don't think the CDN should be the first to do so, unless there's a compelling reason.
2) Will Kibana host frames whose src points to the CDN? If not, then we likely don't need [`frame_src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src).
3) This CDN will (I assume) host static assets. It seems unlikely that [`connect_src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/connect-src) would be required.
4) Will the CDN be hosting scripts used by web workers? If not, then [`worker_src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/worker-src) is unlikely to be required. I don't know where we're using working within Kibana today, so this is a genuine question.